frontend/packages/gitops-plugin/src/components/GitOpsDetailsPage.tsx

```diff
@@ -3,77 +3,47 @@ import * as _ from 'lodash';
+import GitOpsDetails from './details/GitOpsDetails';
 import GitOpsDetailsPageHeading from './details/GitOpsDetailsPageHeading';
-import { GitOpsAppGroupData, GitOpsEnvironment } from './utils/gitops-types';
-import {
-  fetchAppGroups,
-  getEnvData,
-  getPipelinesBaseURI,
-  getApplicationsBaseURI,
-} from './utils/gitops-utils';
+import GitOpsEmptyState from './GitOpsEmptyState';
+import { GitOpsEnvironment } from './utils/gitops-types';
+import { getEnvData, getPipelinesBaseURI, getApplicationsBaseURI } from './utils/gitops-utils';
 import useDefaultSecret from './utils/useDefaultSecret';
+import useEnvDetails from './utils/useEnvDetails';
 
 type GitOpsDetailsPageProps = RouteComponentProps<{ appName?: string }>;
 
 const GitOpsDetailsPage: React.FC<GitOpsDetailsPageProps> = ({ match, location }) => {
   const { t } = useTranslation();
-  const [envs, setEnvs] = React.useState<string[]>(null);
   const [envsData, setEnvsData] = React.useState<GitOpsEnvironment[]>(null);
-  const [emptyStateMsg, setEmptyStateMsg] = React.useState(null);
   const [secretNS, secretName] = useDefaultSecret();
   const { appName } = match.params;
   const searchParams = new URLSearchParams(location.search);
   const manifestURL = searchParams.get('url');
   const pipelinesBaseURI = getPipelinesBaseURI(secretNS, secretName);
   const applicationBaseURI = getApplicationsBaseURI(appName, secretNS, secretName, manifestURL);
   const environmentBaseURI = `/api/gitops/environments`;
-
-  React.useEffect(() => {
-    let ignore = false;
-
-    const getEnvs = async () => {
-      if (!pipelinesBaseURI) return;
-      let appGroups: GitOpsAppGroupData[];
-      let emptyMsg = null;
-      try {
-        appGroups = await fetchAppGroups(pipelinesBaseURI, manifestURL);
-      } catch {} // eslint-disable-line no-empty
-      if (ignore) return;
-      const app = _.find(appGroups, (appObj) => appName === appObj?.name);
-      if (!app?.environments) {
-        emptyMsg = t(
-          'gitops-plugin~Environment details were not found. Try reloading the page or contacting an administrator.',
-        );
-      }
-      setEmptyStateMsg(emptyMsg);
-      setEnvs(app?.environments);
-    };
-
-    getEnvs();
-
-    return () => {
-      ignore = true;
-    };
-  }, [appName, manifestURL, pipelinesBaseURI, t]);
+  const environmentBaseURIV2 = `/api/gitops/environment`;
+  const [envs, emptyStateMsg] = useEnvDetails(appName, manifestURL, pipelinesBaseURI);
 
   React.useEffect(() => {
     const getEnvsData = async () => {
       if (!_.isEmpty(envs) && applicationBaseURI) {
         let data;
         try {
           data = await Promise.all(
-            _.map(envs, (env) => getEnvData(environmentBaseURI, env, applicationBaseURI)),
+            _.map(envs, (env) =>
+              getEnvData(environmentBaseURIV2, environmentBaseURI, env, applicationBaseURI),
+            ),
           );
         } catch {} // eslint-disable-line no-empty
```

Comment:
Another unhandled error catch

Reply:
Added to issue [GITOPS-1278](https://issues.redhat.com/browse/GITOPS-1278)